docs/develop/smart-contracts/guidelines/message-delivery-guarantees.mdx

```diff
@@ -0,0 +1,70 @@
+
+In asynchronous system you can't get response from the destination smart contract in the same transaction. Contract call may take a few blocks to be processed, depending on the length of the route between source and destination.
+
+To achieve the infinite sharding paradigm, it is necessary to ensure full parallelization, which means that the execution of transactions is independent of each other. Therefore, instead of transactions that affect and change the state of many contracts at a time in TON, each transaction is only executed on one smart contract and smart contracts are communicating through messages. That way, smart contracts can only interact with each other by calling their functions with special messages, and getting response from them in other messages later.
+
+If transaction in Ethereum is just a set of function calls in different contracts, transaction in TON is some inbound message that triggered it and some set of outbound messages which are sent to other contracts.
+
+## What is a Logical time?
+
+In such a system with asyncronous and parallel smart contract calls, it can be hard to define the order of actions to process. That's why each message in TON has it's _Logical time_ or _Lamport time_ (later just _lt_). It is used to understand which event caused another one and what do validator need to process first.
+
+It is strictly guaranteed that the transaction resulting from the message will have _lt_ greater than the _lt_ of the message. In order, _lt_ of message sent in some transaction is strictly greater than _lt_ of the transaction. Besides messages sent by one account as well as transactions happened on one account is strictly ordered as well. That way for each account we always know order of transactions, received and send messages.
+
+Moreover if account A sent two messages to account B, it is guaranteed that message with lower _lt_ will be processed earlier. Otherwise, an attempt to synchronize delivery would require the state of all the others to be known before processing one shard, and thereby break parallelization and destroy efficient sharding.
+
+
+For each block, we can define its _lt_ span, which starts with the first transaction and ends with the _lt_ of the last event in the block (message or transaction). Blocks are ordered the same way as any other events in TON, that way if one block depends on the other it has higher _lt_. Child block in shard has higher _lt_ than it's parent. Masterchain block _lt_ is higher that _lt's_ of shard blocks it lists, since master block _depends_ on listed shardblocks. In order each shard block contains reference to latest (at the moment of shard block creation) masterblock and thus shard block _lt_ is higher than referenced master block _lt_.
```

Comment:
For each block, we can define **the** _lt_ span **as starting** **from** the first transaction and **ending** with **the** _lt_ of the last event in the block (message or transaction). Blocks are ordered the same way as other events in TON **and so** if one block depends on the other, it has **a** higher _lt_. **The** **child** block in **a** shard has **a** higher _lt_ than **its** parent. **A** masterchain **block's** _lt_ is higher that the **_lts_** of shard blocks **that** it lists, since **a** master block _depends_ on listed shardblocks. Each shard block contains **ordered** reference to **the** latest (at the moment of shard block creation) masterblock and thus **the** shard block _lt_ is higher than **the** referenced master block _lt_.